### ISSUE TITLE
User not unsubscribed if Admin Newsletter Subscriber page used

### ISSUE THREAD

**Issue by clivewalkden** [klaviyo/magento2-klaviyo#27]
Hi,

This applies to Magento 2.3.1, not tested on anything else.
If you unsubscribe a user from the newsletter subscriber page in the admin they unsubscribe method is never reached.

When the UserProfileNewsletterSubscribeObserver fires the $subscriber->isStatusChanged() always returns false which stops the rest of the execute method running. Magento thinks the user is unsubscribed but the information never makes it back to Klaviyo servers.

Doesn't seem to happen anywhere else on the site (frontend, admin user area, third party modules affecting newsletter subscriptions)

Thanks in advance.

**Comment by clivewalkden** [comment]
Has anyone else experienced this issue?

**Comment by clivewalkden** [comment]
Doesn't seem to syncronize unsubscribes on the Magento side if a user is unsubscribed in the Klaviyo dashboard.

**Comment by remstone7** [comment]
@clivewalkden 

1. We don't post back to the magento side if a user unsubscribes in Klaviyo.  

2. We will investigate the `$subscriber->isStatusChanged()`

**Comment by clivewalkden** [comment]
@remstone7 Thanks for the reply.
1. Why not?
2. Awesome, thanks!

**Comment by remstone7** [comment]
1. great question, it's something i'll pass over to the product managers and see if we can get it in a future release!

**Comment by clivewalkden** [comment]
Awesome, thanks, was hoping there wasn't a major reason!

**Comment by bizlift** [comment]
MailChimp has 2-way sync on subscribe/unsubscribe.  Was testing Klaviyo, and wondering why unsubscribed weren't reflected on Magento. 

Any possibility this functionality will be added so we can migrate to Klaviyo?

**Comment by remstone7** [comment]
@bizlift to confirm you're referring to the admin page specifically (ex. you manually giving someone a status of 0 not 1) and not through the account page?

**Comment by bizlift** [comment]
@remstone7 was referring to comments in thread about Klaviyo > Magento sync direction.  

I landed here via Google & just realized that I should have created a separate issue.  My apologies.

> Doesn't seem to syncronize unsubscribes on the Magento side if a user is unsubscribed in the Klaviyo dashboard.

> We don't post back to the magento side if a user unsubscribes in Klaviyo.

**Comment by clivewalkden** [comment]
@bizlift My original ticket does mention what you are describing so not totally off topic. 2-way sync would be a bonus for this module, would happily give it a + vote if it needs creating as a separate issue.

**Comment by remstone7** [comment]
makes sense, we would do this through out app integration not through the module itself.  I'll open a feature request internally for it

**Comment by bizlift** [comment]
@remstone7 thank you for opening an internal feature request. Any update on whether it's been approved/scheduled?

Is there a temp workaround we could do? Like webhook notification on subscribe/unsubscribe from list?

**Comment by remstone7** [comment]
@bizlift this hasn't been picked up yet.  We unfortunately don't have a webhook based system for this but if you wanted to utilize the metrics api https://www.klaviyo.com/docs/api/metrics#metric-timeline and poll on the unsubscribed from list metric that would work
